### PR TITLE
Change 'mem' namspace to 'jmem'

### PR DESCRIPTION
**Pull request by LaszloLango**
The 'mem_' prefix is too general, so it might clash with
symbols in other libraries. Renamed the directory, file,
funtion and type names.

Related issue: #1052 

JerryScript-DCO-1.0-Signed-off-by: László Langó llango.u-szeged@partner.samsung.com

### CODE REVIEW

File: jerry-core/jmem/jmem-allocator-internal.h [hunk 1]
```diff
@@ -24,7 +24,7 @@
  * @{
  */
 
-extern void mem_run_try_to_give_memory_back_callbacks (mem_try_give_memory_back_severity_t);
+extern void jmem_run_try_to_give_memory_back_callbacks (jmem_try_give_memory_back_severity_t);
```
Comment (zherczeg) [inline jerry-core/jmem/jmem-allocator-internal.h]: The MEM_ALLOCATOR_INTERNAL_H above is not renamed.

Comment (LaszloLango) [inline jerry-core/jmem/jmem-allocator-internal.h]: None of the macro names

Comment (LaszloLango) [inline jerry-core/jmem/jmem-allocator-internal.h]: Fixed

File: jerry-core/jmem/jmem-allocator.c [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+{
+  if (jmem_try_give_memory_back_callback != NULL)
+  {
+    jmem_try_give_memory_back_callback (severity);
+  }
+
+  jmem_pools_collect_empty ();
+} /* jmem_run_try_to_give_memory_back_callbacks */
+
+#ifdef MEM_STATS
```
Comment (zherczeg) [inline jerry-core/jmem/jmem-allocator.c]: I would prefer to rename this to JMEM_STATS. A simple regex can be used to detect all MEM_ prefixes /\bMEM_\B/ and these should be replaced to JMEM_

File: jerry-core/jmem/jmem-heap.c [hunk 3]
```diff
@@ -0,0 +1,775 @@
...
+ */
+
+#include "jrt.h"
+#include "jrt-bit-fields.h"
+#include "jrt-libc-includes.h"
+#include "jmem-allocator.h"
+#include "jmem-config.h"
+#include "jmem-heap.h"
+
+#define JMEM_ALLOCATOR_INTERNAL
```
Comment (zherczeg) [inline jerry-core/jmem/jmem-heap.c]: E.g. this is renamed here.

File: tests/unit/test-poolman.c [hunk 4]
```diff
@@ -61,29 +61,29 @@ main (int __attr_unused___ argc,
       }
     }
 
-    // mem_heap_print (false);
+    // jmem_heap_print (false);
```
Comment (zherczeg) [inline tests/unit/test-poolman.c]: I think such comments should be fixed as well.

Comment (LaszloLango) [inline tests/unit/test-poolman.c]: How?

Comment (zherczeg) [inline tests/unit/test-poolman.c]: /**/

File: jerry-core/jerry.c [hunk 5]
```diff
@@ -1633,23 +1633,23 @@ jerry_init (jerry_flag_t flags) /**< combination of Jerry flags */
 
   if (flags & (JERRY_FLAG_MEM_STATS | JERRY_FLAG_MEM_STATS_SEPARATE))
```
Comment (zherczeg) [inline jerry-core/jerry.c]: at some point the J should be added before these macros as well for consistency but the patch is big enough.

**Comment by zherczeg** [comment]
LGTM

**Comment by akiss77** [comment]
Truth be told, this is a terrible change. But it's needed as a quick fix.
LGTM
But keep the original issue open and we should find a better approach there that can be consistently applied to the whole code base. (As prefixing everything with a "j" must not be the way to go.)

**Pull request merged**